ships/0006-surface-results-in-buildruns.md

```diff
@@ -0,0 +1,293 @@
+- Override the .status.conditions with Tekton results.
+
+## Proposal
+
+A high level proposal will require three things:
+
+- [1] Define the API changes for the BuildRun CRD, this should only be for the _.status_ subresource. See [API Changes](#api-changes).
+
+- [2] Modify the BuildRun CRD to extend the _.status_ subresource with the proposed changes in [1].
+
+- [3] Modify the BuildRun controller business logic, to be able to propagate the Tekton _.status.taskResults_ into [2].
+
+### API Changes
+
+The API changes require a categorization of two types of results. To provide a clear distinction, we will classify them as follows:
+
+- **Strategy Results:** Results that are intended for surfacing higher-level information of a strategy. Is the Strategy Author responsibility to define and emit this results, if desired.
+
+- **Build Results:** Results that are generated at runtime by the BuildRun controller. These results are expected to be present all the time, as long as the step that defines them, exist.
+
+### Strategy Results API Proposal
+
+This will require to introduce `results` under the _.status_ subresource of a BuildRun. This is a list of **name/value**, where Strategy Authors can surface their desire result. This new API change for the BuildRun will be looking as follows:
+
+```yaml
+ status:
+   results:
+   - name: a-strategy-result
+     value: a-value
+   - name: another-strategy-result
+     value: another-value
+```
+
+Additionally, we will need to allow Strategy Authors to define a list of results in the strategy definition. This new API change in the Strategies will be looking as follows:
+
+```yaml
+spec:
+  results:
+    - name: a-strategy-result
+      description: a-meaningful-description
+    - name: another-strategy-result
+      description: a-meaningful-description
+  buildSteps:
+    ...
+
+```
+
+### Build Results
+
+These are results defined at runtime by the BuildRun controller. In this proposal we categorize them in two:
+
+- sources
+- output
+
+Following what was mentioned in [EP removal of tekton resources](https://github.com/shipwright-io/build/blob/main/docs/proposals/removal-tekton-resources.md#3-tekton-results), we need to consider the following:
+
+#### Current State of Build Results for Sources
+
+Assuming the following Build definition:
+
+```yaml
+apiVersion: shipwright.io/v1alpha1
+kind: Build
+metadata:
+  name: a-build
+spec:
+  source:
+    url: https://github.com/shipwright-io/sample-go
+
+```
+
+We will have the following:
+
+| Result Name                       | Naming Convention                      | Emitted Today | Definition  | Notes                                |
+| --------------------------------: | ------------------------------------------: | -------: | ----------: | -----------------------------------: |
+| shp-source-default-commit-sha     |  `shp-source-${sourceType}-${resultName}`   | yes     |  at runtime | `${sourceType}` is `default` as is the only supported git source today |
```

Comment:
For consistency, I would call it name here because the type would be `git`.

```suggestion
| shp-source-default-commit-sha     |  `shp-source-${sourceName}-${resultName}`   | yes     |  at runtime | `${sourceName}` is `default` as `spec.source` has no name. |
```

You can refer to https://github.com/shipwright-io/build/blob/main/docs/proposals/removal-tekton-resources.md#3-tekton-results for the naming logic.